### PR TITLE
Option to import created image directly into local docker registry

### LINKED ISSUE #18: Enhancement: Build to local registry
An option to build to the local image registry similar to jib's `dockerBuild`.

### PR DESCRIPTION
**Pull request by vehagn**
Quick and dirty Docker-integration using `docker` commands to import the tarball created by the tarExporter.

Resolves #18

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -189,9 +191,9 @@ if (options.token) {
 exitWithErrorIf(!options.folder, "--folder must be specified");
 exitWithErrorIf(!options.fromImage, "--fromImage must be specified");
 exitWithErrorIf(!options.toImage, "--toImage must be specified");
-exitWithErrorIf(!options.toRegistry && !options.toTar, "Must specify either --toTar or --toRegistry");
+exitWithErrorIf(!options.toRegistry && !options.toTar && !options.toDocker, "Must specify either --toTar, --toRegistry or --toDocker");
 exitWithErrorIf(
-	!options.toRegistry && !options.toToken && !options.toTar,
+	!options.toRegistry && !options.toToken && !options.toTar &&!options.toDocker,
```
Comment (vehagn) [inline src/cli.ts:196]: issue: When building to a tarball and an external registry the token-check is "ignored" here. 
suggestion: I think `!!options.toRegistry && !options.toToken` would solve this.

File: src/tarExporter.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ async function saveToTar(fromdir: string, tmpdir: string, toPath: string, repoTa
 	];
 	await fs.writeFile(path.join(tardir, "manifest.json"), JSON.stringify(simpleManifest));
 	await fs.writeFile(path.join(tardir, "config.json"), JSON.stringify(config));
-	await tar.c(
+	await tar.create(
```
Comment (vehagn) [inline src/tarExporter.ts:46]: suggestion: From what I understand `c` is just an alias for `create` which describes to operation better imo. `tar.c(...)` was confusing for me to begin with at least.

**Comment by eoftedal** [comment]
Sorry for the late response here. I'm not super keen on making functionality depend on having docker installed, as one of the goals is to have this tool work without docker. Maybe we can figure out how to push it to the docker socket?

**Comment by vehagn** [comment]
I agree that pushing directly to the docker socket is a more elegant solution than depending on having `docker` available. 

My motivation for implementing this functionality is to give the developer an easy way of creating and testing the image locally before sending the build off to the pipeline.

[Looking at the jib source code](https://github.com/GoogleContainerTools/jib/blob/master/jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java#L747-L753) it seems that jib relies on having the `docker` command available, at least for the `mvn compile jib:dockerBuild` command.

From an end-user standpoint I don't think there's many who would want to push to the docker socket without also having docker installed. Maybe a remote socket, but then I would assume it easier to just push to a registry which is already possible without `docker` available. I could be wrong here though.

**Comment by eoftedal** [comment]
Ok, if you can rebase this to head, I will merge it.

**Pull request merged**